Release notes for 8.10.3 (#15390)
elastic/logstash: docs/static/releasenotes.asciidoc

* <<logstash-8-0-0-alpha1,Logstash 8.0.0-alpha1>>

[[logstash-8-10-3]]
=== Logstash 8.10.3 Release Notes

[[known-issues-8.10.3]]
==== Known issues

> Following plugins may fail in Logstash 8.10.3:

Reviewer: ```suggestion
These plugins may fail in Logstash 8.10.3:
```

*Useragent Filter - 3.3.5*

* Upgrade `snakeyaml` dependency https://github.com/logstash-plugins/logstash-filter-useragent/pull/89[#89]

*Beats Input - 6.6.4*

* [DOC] Fix misleading `enrich/source_data` input beats documentation about the Logstash host. https://github.com/logstash-plugins/logstash-input-beats/pull/478[#478]

*Elastic_serverless_forwarder Input - 0.1.3*

* Deprecates the `ssl` option in favor of `ssl_enabled` https://github.com/logstash-plugins/logstash-input-elastic_serverless_forwarder/pull/6[#6]
* Bumps `logstash-input-http` gem version to `>= 3.7.2` (SSL-normalized)

*Aws Integration - 7.1.6*

* Clean up plugin created temporary dirs at startup https://github.com/logstash-plugins/logstash-integration-aws/pull/39[#39]

*Jdbc Integration - 5.4.5*

* Pin sequel to < 5.72.0 due to ruby/bigdecimal#169 https://github.com/logstash-plugins/logstash-integration-jdbc/pull/141[#141]

*Kafka Integration - 11.3.1*

* Fix: update snappy dependency https://github.com/logstash-plugins/logstash-integration-kafka/pull/148[#148]

[[logstash-8-10-2]]
=== Logstash 8.10.2 Release Notes

[[known-issues-8.10.2]]
==== Known issues

> Following plugins may fail in Logstash 8.10.2:

Reviewer: ```suggestion
These plugins may fail in Logstash 8.10.2:
```

*Imap input plugin*

** Due to JRuby upgrade, the plugin is broken and will be unbundled. Details and updates are available in https://github.com/logstash-plugins/logstash-input-imap/issues/61[GitHub issue #61].

*Email output plugin*

** Plugin raises `LoadError: no such file to load -- net/smtp` runtime error. See the issue details and work around in https://github.com/logstash-plugins/logstash-output-email/issues/68[GitHub issue #68].

[[logstash-8-10-1]]
=== Logstash 8.10.1 Release Notes

[[known-issues-8.10.1]]
==== Known issues

> Following plugins may fail in Logstash 8.10.1:

Reviewer: ```suggestion
These plugins may fail in Logstash 8.10.1:
```

*Imap input plugin*

** Due to JRuby upgrade, the plugin is broken and will be unbundled. Details and updates are available in https://github.com/logstash-plugins/logstash-input-imap/issues/61[GitHub issue #61].

*Email output plugin*

** Plugin raises `LoadError: no such file to load -- net/smtp` runtime error. See the issue details and work around in https://github.com/logstash-plugins/logstash-output-email/issues/68[GitHub issue #68].

[[logstash-8-10-0]]
=== Logstash 8.10.0 Release Notes

[[known-issues-8.10.0]]
==== Known issues

> Following plugins may fail in Logstash 8.10.0:

Reviewer: ```suggestion
These plugins may fail in Logstash 8.10.0:
```

** `java.nio.file.NoSuchFileException` when finalizing the segment https://github.com/elastic/logstash/pull/15233[#15233]
** DLQ file using wrong sort order https://github.com/elastic/logstash/pull/15246[#15246]

> ==== Notable plugin issues

Reviewer 2: WDYT @roaksoax for this change?
Noting plugin issues and forwarding to the details would give us a flexibility to update the description when we bundle `output-email` plugin new version and unbundle the `input-imap`.

Reviewer: Also noting that these notes  (Lines 104-113) were added to the 8.10.2 (not 8.10.3) release notes section.  Was that your intention?

Reviewer 2: It is intentionally placed in 8.10.0 section. We came to know recently but plugins started having an issue after 8.10.0

Reviewer: I'm interested to hear what @roaksoax says, but this info seems like it belongs in the 8.10.3 section as "Known  issues."  You can probably find an example if you search source.  When we have Known Issues, they are added as the first section 

We shouldn't expect users looking at the 8.10.3 release notes to know that they should go to the 8.10.0 release notes to see if we've added anything.

> ==== Notable plugin issues

Reviewer: I'm uncomfortable with the "Notable plugin issues" section title.  With 200+ plugins, we're talking on a lot of added responsibility by deciding what is considered "notable."

Reviewer: It seems like it would be easier for users to scan the full list, pick out the plugins they care about, and then see the changes for those.

Reviewer 2: It is my thought and I don't know such situations handled in the past. The intention here, to let customer know mentioned plugins have an issue and bring them to the issue details and solution through the linked issue under the plugin repo. Feel free to suggest the better strategy suitable for this case. Thanks.

Reviewer: After better understanding your intent, I believe we should add this info to the top of 8.10.3 release notes as "Known issues."

Reviewer: We should add the Known Issues to top of 8.10.2, 8.10,1, and 8.10.0 release notes.

Reviewer 2: I have added know imap and email plugin issues to top of 8.10+ sections.
Please review them and if it requires more effort, I am happy to merge without known issues, and make it happen them with separate PR.
Thank you @karenzone !

*Imap input plugin*

> ** Due to JRuby upgrade, the plugin is broken and it will be unbundled. See the issue details https://github.com/logstash-plugins/logstash-input-imap/issues/61[here].

Reviewer: ```suggestion
** Due to JRuby upgrade, the plugin is broken and will be unbundled. Details and updates are available in https://github.com/logstash-plugins/logstash-input-imap/issues/61[GitHub issue #61].
```

*Email output plugin*

> ** Plugin raises `LoadError: no such file to load -- net/smtp` runtime error. See the issue details and solution https://github.com/logstash-plugins/logstash-output-email/issues/68[here].

Reviewer: ```suggestion
** Plugin raises `LoadError: no such file to load -- net/smtp` runtime error. See the issue details and work around in https://github.com/logstash-plugins/logstash-output-email/issues/68[GitHub issue #68].
```

Reviewer: We always make link text as descriptive as possible to align with best practices for SEO and ADA compliance. I can explain why if you'd like more details.
